irr/src/CMakeLists.txt

```
@@ -129,8 +130,7 @@ if(NOT (BSD OR SOLARIS OR EMSCRIPTEN))
 endif()
 
 # OpenGL
-
-if(USE_SDL2)
+if(USE_SDL2 AND NOT ANDROID)
 	option(ENABLE_OPENGL3 "Enable OpenGL 3+" TRUE)
```

I think you meant to change the default
```suggestion
if(USE_SDL2)
	option(ENABLE_OPENGL3 "Enable OpenGL 3+" (NOT ANDROID))
```

---

With the OpenGL 3 driver enabled, it doesn't compile.

```
Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OPENGL_glx_LIBRARY OPENGL_INCLUDE_DIR)
```

As far as I know, Android in general only supports OpenGL ES. (The unified OpenGL ES 2 driver is behind ENABLE_GLES2, not ENABLE_OPENGL3.)

---

yup, no full GL support on Android, only GL ES.

---

The NVIDIA Shield supports desktop GL.
And if Irrlicht is searching for the GL library when you enable OPENGL3 (not OPENGL), then that's a bug. It's not even needed.

---

Ok, you can now compile for Android with OpenGL 3, but it's disabled by default. It doesn't crash at least, enabling it results in this:

```
VERBOSE[Main]: Trying video driver OpenGL 3+
initialize(true, false)
INFO[Main]: Irrlicht: SDL initialized
WARNING[Main]: Irrlicht: Could not create window...: Could not initialize OpenGL / GLES library
ERROR[Main]: Irrlicht: Could not initialize display: Could not initialize OpenGL / GLES library
[ALSOFT] (WW) pthread_setschedparam failed: Operation not permitted (1)
[ALSOFT] (WW) D-Bus not supported
INFO[Main]: Irrlicht: Quit SDL
ERROR[Main]: Failed to initialize the OpenGL 3+ video driver
VERBOSE[Main]: Trying video driver OpenGL ES2
INFO[Main]: Irrlicht: SDL initialized
HostConnection::get() New Host Connection established 0xebcb3700, tid 6018
HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_sync_buffer_data ANDROID_EMU_vulkan_async_qsri ANDROID_EMU_read_color_buffer_dma GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_0 
surfaceChanged()
Window size: 2712x1260
Device size: 2992x1344
pixel format wanted SDL_PIXELFORMAT_RGBX8888 (2), got SDL_PIXELFORMAT_RGBX8888 (2)
eglCreateContext: 0xebcb2d60: maj 3 min 0 rcv 3
eglMakeCurrent: 0xebcb2d60: ver 3 0 (tinfo 0xcac09ad0) (first time)
INFO[Main]: Irrlicht: SDL Version 2.30.1
INFO[Main]: Irrlicht: Using COpenGLES2Driver
Davey! duration=724ms; Flags=1, IntendedVsync=280173506762, Vsync=280290173424, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=280306317489, AnimationStart=280306320379, PerformTraversalsStart=280306322019, DrawStart=280875095756, SyncQueued=280875328201, SyncStart=280875556577, IssueDrawCommandsStart=280875622085, SwapBuffers=280876049749, FrameCompleted=280898260427, DequeueBufferDuration=21494568, QueueBufferDuration=162384, GpuCompleted=0, 
called unimplemented OpenGL ES API
INFO[Main]: Irrlicht: Using renderer: OpenGL ES 3.0 (4.5.0 NVIDIA 550.67)
INFO[Main]: Irrlicht: Vendor: Google (NVIDIA Corporation)
surfaceChanged()
Window size: 2992x1344
Device size: 2992x1344
onWindowFocusChanged(): true
nativeFocusChanged()
INFO[Main]: Irrlicht: Loaded 382 bytes for vertex shader Renderer2D.vsh
INFO[Main]: Irrlicht: Loaded 334 bytes for fragment shader Renderer2D.fsh
INFO[Main]: Irrlicht: Loaded 382 bytes for vertex shader Renderer2D.vsh
INFO[Main]: Irrlicht: Loaded 129 bytes for fragment shader Renderer2D_noTex.fsh
VERBOSE[Main]: Using the OpenGL ES2 video driver
```

I didn't apply your exact suggestion since it resulted in a CMake syntax error.